Fix centre and square state and add rotation for rectangle selector (#20839)
matplotlib/matplotlib: doc/api/next_api_changes/deprecations/20839-EP.rst

Selector widget state internals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> *state_modifier_keys* now have to be defined when creating a selector widget.

Reviewer: ```suggestion
*state_modifier_keys* cannot now be modified after creating a widget, and must be defined when creating the widget.
```

Reviewer 2: ```suggestion
*state_modifier_keys* are immutable now. They can be set when creating the widget, but cannot be changed afterwards anymore.
```

> *state_modifier_keys* have to be defined when creating the selector widget. The 

Reviewer: ```suggestion
*state_modifier_keys* now have to be defined when creating a selector widget. The 
```

> The *state_modifier_keys* attribute has been privatized and deprecated. Use
> *state_modifier_keys* argument when creating the selector instead.

Reviewer 2: ```suggestion
*state_modifier_keys* have to be defined when creating the selector widget. The 
*state_modifier_keys* attribute is deprecated.
```

Author: Done.
